addons/website/controllers/main.py

```diff
@@ -208,10 +206,10 @@ def reset_template(self, templates, redirect='/'):
                 modules_to_update.append(view.model_data_id.module)
 
         if modules_to_update:
-            module_obj = request.registry['ir.module.module']
-            module_ids = module_obj.search(request.cr, request.uid, [('name', 'in', modules_to_update)], context=request.context)
+            IrModule = request.env['ir.module.module']
+            module_ids = IrModule.search([('name', 'in', modules_to_update)]).ids
             if module_ids:
-                module_obj.button_immediate_upgrade(request.cr, request.uid, module_ids, context=request.context)
+                IrModule.button_immediate_upgrade(module_ids)
```

Comment:
The method `button_immediate_upgrade` should be invoked on a recordset (with the `ids` argument in the recordset). Make it simpler by using recordset and not getting `ids` out of them:

    IrModule = request.env['ir.module.module']
    modules = IrModule.search([('name', 'in', modules_to_update)])
    if modules:
        modules.button_immediate_upgrade()